### PR TITLE
Fix fit method

### PR DESCRIPTION
**Pull request by zhichao-li**

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala [hunk 1]
```diff
@@ -700,3 +740,27 @@ object Sequential extends KerasLayerSerializable {
     new Sequential[T]()
   }
 }
+
+private[zoo] class ZooLocalOptimizer[T: ClassTag] (
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala]: InternalLocalOptimizer?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala [hunk 2]
```diff
@@ -700,3 +740,27 @@ object Sequential extends KerasLayerSerializable {
...
+    model: Module[T],
+    ds: LocalDataSet[MiniBatch[T]],
+    criterion: Criterion[T]
+)(implicit ev: TensorNumeric[T]) extends LocalOptimizer[T](model, ds, criterion) {
+
+  def setTrainData(trainingDataSet: DataSet[MiniBatch[T]]): this.type = {
+     this.dataset = trainingDataSet
+    this
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala]: `endEpoch()`?

Comment (zhichao-li) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala]: `optimize()` is a blocking method. Seems like we don't need to manually invoke `endEpoch` for increasing the epoch number. And also if we invoke `setTrainData` multiple times then the `endEpoch` would be increased even if we haven't started to `optimize()` cc @yangw1234 what's the reason for adding `endEpoch`?

Comment (yangw1234) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala]: @zhichao-li endEpoch only increase epoch number when the current epoch is not finished. If current epoch is finished, endEpoch will do nothing.

**Comment by zhichao-li** [comment]
Jenkins passed. #829

**Pull request merged**